sklearn/tests/test_common.py

```diff
@@ -444,3 +445,21 @@ def test_estimators_do_not_raise_errors_in_init_or_set_params(Estimator):
 
         # Also do does not raise
         est.set_params(**new_params)
+
+
+VALIDATE_REGRESSOR_DTYPE = [
+    "LinearRegression",
+    "DecisionTreeRegressor",
+    "SVR",
+    "GradientBoostingRegressor",
+    "RandomForestRegressor",
+]
+VALIDATE_REGRESSOR_DTYPE = set(VALIDATE_REGRESSOR_DTYPE)
+
+
+@pytest.mark.parametrize(
+    "Regressor",
+    [est for name, est in all_estimators() if name in VALIDATE_REGRESSOR_DTYPE],
+)
+def test_regressor_preserve_dtypes(Regressor):
+    check_regressor_preserve_dtypes(Regressor.__name__, Regressor)
```

Comment:
Instead of writing a new test function in `test_common.py` with a list of hard-coded regressor names, you should instead call the `check_regressor_preserve_dtypes` in `_yield_regressor_checks` of `sklearn/utils/estimator_checks.py` with something like:


```python
def _yield_regressor_checks(regressor):
    tags = _safe_tags(regressor)
    ...
    if tags["preserves_dtype"]:
        yield check_regressor_preserve_dtypes
    ...
```

similarly to what is done in the `_yield_transformer_checks` function for the call to `check_transformer_preserve_dtypes`.